itur/models/itu1623.py

```diff
@@ -0,0 +1,668 @@
+        p = np.zeros_like(D_arr)  # initializes p for indexing ops.
+        Q_ratio_p = (Qfunc(np.log(D_arr / D_2) / sigma) /
+                     Qfunc(np.log(D_t / D_2) / sigma))
+
+        p = np.where(
+            np.logical_and(D_arr >= 1, D_arr <= D_t),
+            D_arr ** -gamma,
+            (D_t ** -gamma) * Q_ratio_p,
+        )
+
+        # Step 8: Calculate the cumulative probability of exceedance, i.e. the
+        # total fraction of fade time due to fades of duration d longer than D:
+        F = np.zeros_like(D_arr)
+        Q_ratio_F = (Qfunc(np.log(D_arr / D_0) / sigma) /
+                     Qfunc(np.log(D_t / D_0) / sigma))  # or divide by Q_2
+
+        F = np.where(
+            np.logical_and(D_arr >= 1, D_arr <= D_t),
+            (1 - (k * (D_arr / D_t) ** (1 - gamma))),
+            ((1 - k) * Q_ratio_F),
+        )
+
+        # Step 9: Compute N(D,A), The total number of fades of duration d
+        # longer than D for a given threshold A.
+
+        # Step 9a. Compute Ntot, using  the Ttot(A) parameter
+        N_tot = T_tot * (k / gamma) * ((1 - gamma) / (D_t ** (1 - gamma)))
+
+        # Compute number of fades N(D,A)
+        N = N_tot * p
+
+        # Compute T(d > D|a > A), total fading time due to fades of duration d
+        # longer than D for the threshold A:
+        T = T_tot * F
+
+        return np.array([p, F, N, T])
+
+    @classmethod
+    def fade_slope(self, z, A, f_B, delta_t):
+        # Step 1:   Calculate F
+        b = 2.3
+        F = np.sqrt(
+            (2 * np.pi) ** 2 / ((1 / f_B ** b) + (2 * delta_t) ** b) ** (1 / b)
+        )   # eq. 18
+
+        # Step 2:   Calculate STD of the conditional fade slope
+        # s is a parameter which depends on climate and elevation angle; an
+        # overall average value in Europe and the United States of America,
+        # at elevations between 10° and 50°, is s = 0.01
+        s = 0.01
+        sigma_z = s * F * A
+
+        # Step 3a:  Calculate the conditional probability
+        p = 2 / (np.pi * sigma_z * (1 + (z / (sigma_z) ** 2) ** 2))  # eq. 20
+
+        # Step 3b : If required, calculate p(ζ |A), the conditional
+        # probability (complementary cumulative distribution function) that
+        # the fade slope ζ is exceeded for a given attenuation value, A:
+        z_over_sigmaz = z / sigma_z
+        abs_z_over_sigmaz = np.abs(z) / sigma_z
+        P = (
+            0.5
+            - z_over_sigmaz / (np.pi * (1 + z_over_sigmaz ** 2))
+            - (np.arctan(z_over_sigmaz) / np.pi)
+        )
+
+        # Step 3b2 : calculate p(ζ |A), the conditional probability that the
+        # absolute value of the fade slope ζ is exceeded for a given
+        # attenuation value, A:
+        P2 = (
+            1
+            - 2 * abs_z_over_sigmaz / (np.pi * (1 + abs_z_over_sigmaz ** 2))
+            - (2 * np.arctan(abs_z_over_sigmaz) / np.pi)
+        )
+
+        return p, P, P2, sigma_z
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears you are unpacking more than three variables while using a function that returns more than one variable. Using a large number of return values is prone to errors. We recommend that you return a small class or `namedtuple` instance instead.

[Learn more](https://docs.python.org/3/library/typing.html#typing.NamedTuple)